File: aoj/volume100/python3/A10032.py
import sys

stack = []
for operation in sys.stdin:
    operation = operation.strip()

    if operation.startswith('quit'):
        break

    elif operation.startswith('push'):
        _, block = operation.split()
        stack.append(block)

    elif operation.startswith('pop'):
        print(stack.pop())

File: aoj/volume100/nodejs/A10032.js
function main(lines) {
  var stack = [];
  for (var i = 0;; i++) {
    var operation = lines[i];
    if (operation.match(/^quit/)) {
      break;
    } else if (operation.match(/^push/)) {
      var block = operation.split(/\s+/)[1];
      stack.push(block);
    } else if (operation.match(/^pop/)) {
      var block = stack.pop();
      console.log(block);
    }
  }
}

var chunks = [];
process.stdin.on('data', function(chunk) {
  chunks.push(chunk)
});
process.stdin.on('end', function() {
  var lines = chunks.join('').split('\n');
  main(lines);
});
